packages/util-endpoints/src/cache/EndpointCache.ts

```diff
@@ -0,0 +1,62 @@
+import type { EndpointParams, EndpointV2 } from "@smithy/types";
+
+/**
+ * @internal
+ *
+ * Cache for endpoint ruleSet resolution.
+ */
+export class EndpointCache {
+  private capacity: number = 50;
+  private data = new Map<string, EndpointV2>();
+  private parameters: string[] = [];
+
+  /**
+   * @param [params] - list of params to consider as part of the cache key.
+   *
+   * If the params list is not populated, all object keys will be considered.
+   * This may be out of order depending on how the object is created and arrives to this class.
+   */
+  public constructor({ size, params }: { size?: number; params?: string[] }) {
+    this.capacity = size ?? 50;
```

Comment:
nit: This is already initialized as `50`on line 9, and doesn't have to be overridden
```suggestion
    if (size) {
      this.capacity = size;
    }
```